ansible/roles/compute_init/files/compute-init.yml

```diff
@@ -0,0 +1,271 @@
+            state: directory
+            owner: "{{ item.mount_user | default(omit) }}"
+            group: "{{ item.mount_group | default(omit) }}"
+            mode: "{{ item.mount_mode | default(omit) }}"
+          loop: "{{ os_manila_mount_shares }}"
+          loop_control:
+            label: "{{ item.share_name }}"
+          when: item.mount_state | default(os_manila_mount_state) in ['mounted' or 'ephemeral']
+
+
+    - name: Basic users setup
+      block:
+        - name: Create groups
+          ansible.builtin.group: "{{ item }}"
+          loop:  "{{ basic_users_groups }}"
+
+        - name: Create users
+          user: "{{ basic_users_userdefaults | combine(item) | filter_user_params() }}"
+          loop: "{{ basic_users_users }}"
+          loop_control:
+            label: "{{ item.name }} [{{ item.state | default('present') }}]"
+          register: basic_users_info
+
+        - name: Write sudo rules
+          blockinfile:
+            path: /etc/sudoers.d/80-{{ item.name}}-user
+            block: "{{ item.sudo }}"
+            create: true
+          loop: "{{ basic_users_users }}"
+          loop_control:
+            label: "{{ item.name }}"
+          when: "'sudo' in item"
+
+
+    - name: Configure EESSI
+      block:
+        - name: Copy /mnt/cluster/cvmfs/default.local contents to /etc/cvmfs/default.local
+          copy:
+            src: /mnt/cluster/cvmfs/default.local
+            dest: /etc/cvmfs/default.local
+            owner: root
+            group: root
+            mode: 0644
+
+        # NOTE: Not clear how to make this idempotent
+        - name: Ensure CVMFS config is setup
+          command:
+            cmd: "cvmfs_config setup"
+
+
+    - name: Configure openhpc
+      block:
+        - name: Fix permissions on /etc to pass Munge startup checks
+          # Rocky-9-GenericCloud-Base-9.4-20240523.0.x86_64.qcow2 makes /etc g=rwx rather than g=rx (where group=root)
+          # which fails munged startup checks
+          file:
+            path: /etc
+            state: directory
+            mode: g-w
+
+        - name: Copy Munge key from NFS-mounted directory to /etc/munge
+          copy:
+            src: "/mnt/cluster/openhpc_munge.key"
+            dest: "/etc/munge/munge.key"
+            owner: munge
+            group: munge
+            mode: 0400
+
+        - name: Set slurmctld location for configless operation
+          lineinfile:
+            path: /etc/sysconfig/slurmd
+            line: "SLURMD_OPTIONS='--conf-server {{ openhpc_conf_server }}'"
+            regexp: "^SLURMD_OPTIONS="
+            create: yes
+            owner: root
+            group: root
+            mode: 0644
+
+        - name: Configure Munge service
+          service:
+            name: munge
+            enabled: true
+            state: started
+
+        - name: Ensure slurmd state
+          service:
+            name: slurmd
+            enabled: true
+            state: started
```

Comment:
Need to ensure the node is resumed, because slurmctld will think this has "unexpectedly rebooted"

something like:
```
- name: Ensure node is in cluster
  command: scontrol update state=resume nodename=<hostname>
```

I'm not sure how you get hostname best - normally it'd be `{{ inventory_hostname }}` but that won't work here. It needs to be the short name. Maybe using `={{ lookup('pipe', 'hostname -s') }}` is not mad?